internal/controllers/cluster/cluster_controller.go

```diff
@@ -215,6 +216,16 @@ func (r *Reconciler) reconcile(ctx context.Context, cluster *clusterv1.Cluster)
 func (r *Reconciler) reconcileDelete(ctx context.Context, cluster *clusterv1.Cluster) (reconcile.Result, error) {
 	log := ctrl.LoggerFrom(ctx)
 
+	// If the RuntimeSDK and ClusterTopology flags are enabled, for clusters with managed topologies
+	// only proceed with delete if the cluster is marked as `ok-to-delete`
+	if feature.Gates.Enabled(feature.RuntimeSDK) && feature.Gates.Enabled(feature.ClusterTopology) {
+		if cluster.Spec.Topology != nil {
+			if !hooks.IsOkToDelete(cluster) {
+				return ctrl.Result{}, nil
+			}
+		}
+	}
```

Comment:
```suggestion
		if cluster.Spec.Topology != nil && !hooks.IsOkToDelete(cluster) {
			return ctrl.Result{}, nil
		}
	}
```
nit: just to flatten it a bit